### DISCUSSION TITLE
Is it guaranteed for GcHandle.ToIntPtr not to return zero for non-null values?

### DISCUSSION THREAD

**Discussion by anatawa12** [dotnet/runtime#96958] [Q&A] [answered]
I could not find documentation that express this but is it guaranteed?
If so, could you document?

**Comment by huoyaoyuan** [comment]
https://github.com/dotnet/runtime/blob/300013b6dffdfeb1864ad5a6a929c75f9278fe50/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/GCHandle.cs#L146

It's somehow a common contract of "handle": 0 value always means equivalent of null.

**Reply by anatawa12** [reply]
I found similar codes in the GCHandle struct but I could not believe it's not a implementation details.

**Reply by huoyaoyuan** [reply, marked as answer]
In my mind, it's too common beyond GCHandle. It's applicable for any handle.
